code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -259,12 +261,12 @@
 	max_share = 0 //Gotta reset our tracker
 	#endif
 
-	for(var/t in adjacent_turfs)
-		var/turf/open/enemy_tile = t
-
+	for(var/turf/open/enemy_tile as anything in adjacent_turfs)
 		if(fire_count <= enemy_tile.current_cycle)
 			continue
-		enemy_tile.archive()
+		enemy_tile.air.archive()
+		enemy_tile.archived_cycle = SSair.times_fired
+		enemy_tile.temperature_archived = enemy_tile.temperature
```

Comment:
Make this a define and use it everywhere this pattern appears, if we're gonna unroll it we should do it properly